core/src/main/java/io/fabric8/maven/core/util/KubernetesResourceUtil.java

```diff
@@ -719,9 +719,20 @@ public static void mergePodSpec(PodSpecBuilder builder, PodSpec defaultPodSpec,
                 int idx = 0;
                 for (Container defaultContainer : defaultContainers) {
                     Container container;
+                    boolean isNewContainerAdded = false;
                     if (idx < containers.size()) {
                         container = containers.get(idx);
                     } else {
+                        isNewContainerAdded = true;
+                        container = new Container();
+                        containers.add(container);
+                    }
+                    //TODO: SK remove it after review
+                    //Though we have more containers in the builder, but if existing one does not match
+                    // default container name then we assume it to be new one or the one to be added
+                    String containerName = container.getImage();
+                    String defaultContainerName = defaultContainer.getImage();
+                    if (!containerName.equalsIgnoreCase(defaultContainerName) && !isNewContainerAdded) {
```

Comment:
Something like that @kameshsampath, but I see that code adds another container if the fragment does not declare a name (that is the standard case when adding fragments). I'd add another container only if the one declared in the fragments uses a name that is different from the default container name for the project (e.g. it's called `proxy`). Please, also fix the dependencies of the 'it' tests as they refer snapshot artifacts.